src/generic.jl

```diff
@@ -212,3 +264,18 @@ function pairwise(metric::PreMetric, a::AbstractMatrix;
     r = Matrix{result_type(metric, a, a)}(undef, n, n)
     pairwise!(r, metric, a, dims=dims)
 end
+
+
+function pairwise(metric::PreMetric, a::AbstractVector, b::AbstractVector)
```

Comment:
```suggestion
function pairwise(metric::PreMetric, a::AbstractVector, b::AbstractVector;
                  dims::Union{Nothing,Integer}=nothing)
    dims = deprecated_dims(dims)
    dims == 2 || throw(ArgumentError("dims must be 2 (got $dims)"))
```

Reply:
I am not sure what is the point of giving a `dims` argument. It is meaningless in this case.

Reply:
Sure, but that's the API, AFAIU. In the matrix case, you even get a warning when you don't provide the `dims` keyword, so it's mandatory. And in the vector case, you included it correctly in `pairwise!`  yourself. I think we should include some `dims` handling there, too. But in any case we need to be consistent.

Reply:
Oh fair point. But in this case I would remove the `dims` in `pairwise!` then. I don't think one needs the exact same API for Vectors and Matrices. This could even be a source of bugs.
Just to refer to KernelFunctions, we provide an API to both but only the Matrix API takes a `dims` argument

Reply:
I see. We should ask for @nalimilan's advice then, because he introduced that mandatory keyword argument for consistency with some other packages from the Statistics ecosystem.

BTW, I also noticed that dims-warning when running the tests from this PR.

Reply:
The idea with the mandatory `dims` argument is that we need to change the default to become consistent with Statistics in the stdlib (e.g. `cor`). So until we do that we have to require people to pass it explicitly so that their code doesn't break the day we make the switch. But here I agree it doesn't make any sense to support it since there are no rows and columns, only vectors of vectors.